src/python/pants/init/extension_loader.py

```diff
@@ -75,58 +79,68 @@ def load_plugins(build_configuration, plugins, working_set):
-      subsystems = entries['global_subsystems'].load()()
-      build_configuration.register_optionables(subsystems)
-
-    if 'rules' in entries:
-      rules = entries['rules'].load()()
-      build_configuration.register_rules(rules)
-
+    if is_v1_plugin:
+      if 'register_goals' in entries:
+        entries['register_goals'].load()()
+
+      # TODO: Might v2 plugins need to register global subsystems? Hopefully not.
+      if 'global_subsystems' in entries:
+        subsystems = entries['global_subsystems'].load()()
+        build_configuration.register_optionables(subsystems)
+
+      # The v2 target API is still TBD, so we keep build_file_aliases as a v1-only thing.
+      # Having thus no overlap between v1 and v2 backend entrypoints makes things much simpler.
+      # TODO: Revisit, ideally with a v2-only entry point, once the v2 target API is a thing.
+      if 'build_file_aliases' in entries:
+        aliases = entries['build_file_aliases'].load()()
+        build_configuration.register_aliases(aliases)
+    else:
+      if 'rules' in entries:
+        rules = entries['rules'].load()()
+        build_configuration.register_rules(rules)
     loaded[dist.as_requirement().key] = dist
 
 
-def load_build_configuration_from_source(build_configuration, backends=None):
+def load_build_configuration_from_source(build_configuration, backends1, backends2):
   """Installs pants backend packages to provide BUILD file symbols and cli goals.
 
   :param BuildConfiguration build_configuration: The BuildConfiguration (for adding aliases).
-  :param backends: An optional list of additional packages to load backends from.
+  :param backends1: An list of packages to load v1 backends from.
+  :param backends2: An list of packages to load v2 backends from.
   :raises: :class:``pants.base.exceptions.BuildConfigurationError`` if there is a problem loading
     the build configuration.
   """
   # pants.build_graph and pants.core_task must always be loaded, and before any other backends.
-  # TODO: Consider replacing the "backend" nomenclature here. pants.build_graph and
-  # pants.core_tasks aren't really backends.
-  backend_packages = OrderedSet([
+  backend_packages1 = OrderedSet([
       'pants.build_graph',
       'pants.core_tasks',
+    ] + (backends1 or []))
+  for backend_package in backend_packages1:
+    load_backend(build_configuration, backend_package, is_v1_backend=True)
+
+  backend_packages2 = OrderedSet([
       'pants.rules.core',
-    ] + (backends or []))
-  for backend_package in backend_packages:
-    load_backend(build_configuration, backend_package)
+    ] + (backends2 or []))
+  for backend_package in backend_packages2:
+    load_backend(build_configuration, backend_package, is_v1_backend=False)
 
 
-def load_backend(build_configuration: BuildConfiguration, backend_package: str) -> None:
+def load_backend(build_configuration: BuildConfiguration, backend_package: str,
+                 is_v1_backend: bool) -> None:
```

Comment:
```suggestion
                 * , is_v1_backend: bool) -> None:
```
This would make it a keyword-only arg, which I think is the intent here (to ensure callers make it clear what type of plugin they're loading). This is highly subjective and can be ignored.